commands/system/help.js

```diff
@@ -9,74 +11,170 @@ class Help extends Command {
+    const currentCategory = "";
+    const last = [];
+    const sorted = this.client.commands.sort((p, c) => (p.help.cat > c.help.category ? 1 : p.help.name > c.help.name && p.help.cat === c.help.category ? 1 : -1));
     if (!type) {
       const description = `Command category list\n\nUse \`${message.settings.prefix}help <category>\` to find commands for a specific category`;
-      const output = sorted.filter(c => !(level < 10 && c.help.category == "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map(c => {
-        const cat = c.help.category.toProperCase();
+      const output = sorted.filter(c => !(level < 10 && c.help.category === "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map((c) => {
+        const cat = c.help.category;
         if (currentCategory !== cat && !type) {
-          currentCategory = cat;
-          return `\n\`${message.settings.prefix}help ${cat.toLowerCase()}\` | Shows ${cat} commands`;
+          if (last.includes(cat.toLowerCase())) return;
+          last.push(cat.toLowerCase());
+          return `\n\`${message.settings.prefix}help ${cat.toLowerCase()}\``;
         }
       }).join("");
-
       embed.setDescription(description)
         .addField("Categories", output);
     } else {
-      let n = 0;
-      sorted.forEach(c => {
+      sorted.forEach((c) => {
         if (c.help.category.toLowerCase() === type.toLowerCase()) {
           n++;
         }
       });
-    
       let output = "";
-      let num = 0;
-      const pg = parseInt(page) && parseInt(page) <= Math.ceil(n / perpage) ? parseInt(page) : 1;
+      const pg = Number(page) && Number(page) <= Math.ceil(n / perpage) ? Number(page) : 1;
       for (const c of sorted.values()) {
         if (c.help.category.toLowerCase() === type.toLowerCase()) {
-          if (c.help.category === "Owner" && level < 10 ) return;
           if (c.help.category === "NSFW" && !message.channel.nsfw) return;
           if (num < perpage * pg && num > perpage * pg - (perpage + 1)) {
             if (level < this.client.levelCache[c.conf.permLevel]) return;
-            output += `\n\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 50 ? c.help.description.slice(0,50) +"...": c.help.description}`;
+            output += `\n\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 80 ? `${c.help.description.slice(0, 80)}...` : c.help.description}`;
           }
           num++;
         }
       }
-    
+
       if (num) {
-        embed.setTitle("Command category help")
-          .setDescription(`A list of commands in the ${type} category.\n(Total of ${num} commands in this category)\n\nTo get help on a specific command do \`${message.settings.prefix}help <command>\`\n\n${num > 10 && pg === 1 ? `To view more commands do\` ${message.settings.prefix}help <category> 2\`` : "" }`)
-          .addField("Commands", output);
+        Math.ceil(num / perpage) === 1 ?
```

Comment:
This ternary here is unneeded, the next 3-6 lines are duplicated. What is changed by this condition is the title.

Reply:
✔️

Reply:
✔️